src/operator/src/statement/cursor.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common_query::{Output, OutputData};
+use common_recordbatch::cursor::RecordBatchStreamCursor;
+use common_recordbatch::RecordBatches;
+use query::parser::QueryStatement;
+use session::context::QueryContextRef;
+use snafu::ResultExt;
+use sql::statements::cursor::{CloseCursor, DeclareCursor, FetchCursor};
+use sql::statements::statement::Statement;
+
+use crate::error::{self, Result};
+use crate::statement::StatementExecutor;
+
+impl StatementExecutor {
+    pub(super) async fn declare_cursor(
```

Comment:
Adds tracing for these functions.
```suggestion
    #[tracing::instrument(skip_all)]
    pub(super) async fn declare_cursor(
```